include/RAJA/policy/tbb/forall.hpp

```diff
@@ -77,15 +73,17 @@ RAJA_INLINE void forall_impl(const tbb_for_dynamic& p,
                              Func&& loop_body)
 {
   using std::begin;
+  using std::distance;
   using std::end;
-  using brange = ::tbb::blocked_range<decltype(iter.begin())>;
-  ::tbb::parallel_for(brange(begin(iter), end(iter), p.grain_size),
+  using brange = ::tbb::blocked_range<size_t>;
+  auto b = begin(iter);
+  ::tbb::parallel_for(brange(0, distance(begin(iter), end(iter)), p.grain_size),
                       [=](const brange& r) {
                         using RAJA::internal::thread_privatize;
                         auto privatizer = thread_privatize(loop_body);
                         auto body = privatizer.get_priv();
-                        for (const auto& i : r)
-                          body(i);
+                        for (auto i = r.begin(); i != r.end(); ++i)
```

Comment:
It's a good thing we're filling in gaps in our testing...

Reply:
I agree, though I'm not sure why you're flagging this line? The original worked for all iterators that could be wrapped with the tbb range we were using, which come to find out assumes uses a size_t difference type internally, so it only works if end is reachable by using `++` on begin enough times, which sometimes isn't the case for us currently since we aren't actually providing STL compatible ranges as it turns out.

Reply:
Because we didn't have any tests before that combined tbb exec policies with range-stride segments.  Perhaps, my intent wasn't clear  -- it was meant to be kudos to you for figuring out this change was needed and a bit of sarcasm directed at the fact that we weren't covering all our use cases in our tests.